chain/sync.go

```diff
@@ -687,14 +694,44 @@ func (syncer *Syncer) VerifyElectionPoStProof(ctx context.Context, h *types.Bloc
 	}
 
 	if build.InsecurePoStValidation {
-		if string(h.EPostProof.Proof) == "valid proof" {
+		if len(h.EPostProof.Proofs) == 0 {
+			return xerrors.Errorf("[TESTING] No election post proof given")
+		}
+
+		if string(h.EPostProof.Proofs[0].ProofBytes) == "valid proof" {
 			return nil
 		}
 		return xerrors.Errorf("[TESTING] election post was invalid")
 	}
-	hvrf := sha256.Sum256(h.EPostProof.PostRand)
 
-	ok, err := sectorbuilder.ProofVerifier.VerifyElectionPost(ctx, ssize, *sectorInfo, hvrf[:], h.EPostProof.Proof, winners, h.Miner)
+	rt, _, err := api.ProofTypeFromSectorSize(ssize)
+	if err != nil {
+		return err
+	}
+
+	candidates := make([]abi.PoStCandidate, 0, len(h.EPostProof.Candidates))
+	for _, c := range h.EPostProof.Candidates {
+		candidates = append(candidates, abi.PoStCandidate{
+			RegisteredProof: rt,
+			PartialTicket:   c.Partial,
+			SectorID:        abi.SectorID{Number: c.SectorID}, // this should not be an ID, we already know who the miner is...
+			ChallengeIndex:  int64(c.ChallengeIndex),
+		})
+	}
+
+	// TODO: why do we need this here?
+	challengeCount := sectorbuilder.ElectionPostChallengeCount(uint64(len(sectorInfo)), 0)
```

Comment:
Isn't the challengeCount here actually just `len(sectorInfo)`?

Reply:
No, I think that this is right, but i'm confused as to why I need to care here now